src/Sentry/Internal/SentryScopeManager.cs

```diff
@@ -94,21 +95,23 @@ public void BindClient(ISentryClient client)
             _options?.DiagnosticLogger?.LogDebug("Binding a new client to the current scope.");
 
             var currentScopeAndClientStack = ScopeAndClientStack;
-            var newScopeAndClientStack = new Stack<KeyValuePair<Scope, ISentryClient>>(currentScopeAndClientStack);
-            var top = newScopeAndClientStack.Pop();
-            newScopeAndClientStack.Push(new KeyValuePair<Scope, ISentryClient>(top.Key, client ?? DisabledHub.Instance));
+            var top = currentScopeAndClientStack[currentScopeAndClientStack.Length - 1];
+
+            var newScopeAndClientStack = new KeyValuePair<Scope, ISentryClient>[currentScopeAndClientStack.Length];
+            Array.Copy(currentScopeAndClientStack, newScopeAndClientStack, currentScopeAndClientStack.Length);
```

Comment:
nit: Consider using `Buffer.BlockCopy`

Reply:
Any reason for this choice ? [StackOverflow seems to prefer Array.Copy](https://stackoverflow.com/questions/1389821/array-copy-vs-buffer-blockcopy)

Reply:
SO "prefers" because the Buffer.BlockCopy is easier to screw up with the parameters. But we'll test/review things correctly here. I prefer it in this case since they do the same and Buffer is slightly faster even though we're not benchmarking anything here so it might be premature optimization but since they do the same thing, I prefer one over the other.
But as I said: `nit` and _consider_. It's your call.

Reply:
Think `Array.Copy` is good enough for me. If something should be changed, then it is double-stack implementation. It seems a little organic to me. But there is probably a good reason for it.